superset/assets/spec/javascripts/sqllab/ResultSet_spec.jsx

```diff
@@ -2,20 +2,114 @@ import React from 'react';
 import { shallow } from 'enzyme';
 import { describe, it } from 'mocha';
 import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { Alert, ProgressBar } from 'react-bootstrap';
 import FilterableTable from '../../../javascripts/components/FilterableTable/FilterableTable';
+import VisualizeModal from '../../../javascripts/SqlLab/components/VisualizeModal';
 import ResultSet from '../../../javascripts/SqlLab/components/ResultSet';
-import { queries } from './fixtures';
+import { queries, stoppedQuery, runningQuery, cachedQuery } from './fixtures';
 
 describe('ResultSet', () => {
+  const clearQuerySpy = sinon.spy();
+  const fetchQuerySpy = sinon.spy();
   const mockedProps = {
+    actions: {
+      clearQueryResults: clearQuerySpy,
+      fetchQueryResults: fetchQuerySpy,
+    },
+    cache: true,
     query: queries[0],
     height: 0,
   };
+  const stoppedQueryProps = Object.assign({}, mockedProps, {
+    query: stoppedQuery,
+  });
+  const runningQueryProps = Object.assign({}, mockedProps, {
+    query: runningQuery,
+  });
+  const cachedQueryProps = Object.assign({}, mockedProps, {
+    query: cachedQuery,
+  });
+  const newProps = {
+    query: {
+      cached: false,
+      resultsKey: 'new key',
+      results: {
+        data: [{ a: 1 }],
+      },
+    },
+  };
+
   it('is valid', () => {
     expect(React.isValidElement(<ResultSet {...mockedProps} />)).to.equal(true);
   });
   it('renders a Table', () => {
     const wrapper = shallow(<ResultSet {...mockedProps} />);
     expect(wrapper.find(FilterableTable)).to.have.length(1);
   });
+  describe('getControls', () => {
```

Comment:
this could be an `it` block rather than describe. typically describe would wrap multiple `it` blocks.

might be more clear as `it('sets correct class name given props', () => {});`